.github/workflows/build-wheels.yml

```diff
@@ -0,0 +1,101 @@
+name: Build wheels
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * 0,3' # 2/weekly
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        os: [ubuntu-latest, macos-latest]
+        platform: [x64]
+        PLAT: [x86_64]
+    env:
+      PKG_NAME: gensim
+      BUILD_COMMIT: 4.0.0beta
+      UNICODE_WIDTH: 32
+      MB_PYTHON_VERSION: ${{ matrix.python-version }}
+      TRAVIS_PYTHON_VERSION: ${{ matrix.python-version }}
+      TRAVIS_BRANCH: ${{ github.head_ref }}
+      TRAVIS_PULL_REQUEST: ${{ github.event.number }}
+      TRAVIS_BUILD_DIR: ${{ github.workspace }}
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Pin Morfessor, Levenshtein, visdom
+      run: |
+        echo "MF_TEST_DEP=$(echo Morfessor==2.0.2a4)" >> $GITHUB_ENV;
+        echo "LV_TEST_DEP=$(echo python-levenshtein==0.12.0)" >> $GITHUB_ENV;
+        echo "VD_TEST_DEP=$(echo visdom==0.1.8.9)" >> $GITHUB_ENV
+    - name: Setup Environment variables
+      run: |
+        echo "Event type: ${{ github.event_name }}"
+        if [ "schedule" == "${{ github.event_name }}" ]; then
+          echo "TRAVIS_EVENT_TYPE=cron" >> $GITHUB_ENV;
+          echo "BUILD_COMMIT=master" >> $GITHUB_ENV
+        else
+          echo "TRAVIS_EVENT_TYPE=${{ github.event_name }}" >> $GITHUB_ENV
+          echo "BUILD_COMMIT=$BUILD_COMMIT" >> $GITHUB_ENV
+        fi
+        if [ "macos-latest" == "${{ matrix.os }}" ]; then
+          echo "TRAVIS_OS_NAME=osx" >> $GITHUB_ENV
+        else
+          echo "TRAVIS_OS_NAME=${{ matrix.os }}" >> $GITHUB_ENV;
+        fi
+        if [ "${{ matrix.python-version }} != 3.6" && "ubuntu-latest == ${{ matrix.os }}" ]; then
```

Comment:
I think you're over-depending on bash conditionals here. You're mixing actual build steps and configuration, so it's difficult to understand what's going on without reading through the entire workflow.

Isn't it possible to define the environment variables as part of the matrix? For example:

- [add combinations explicitly](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-including-new-combinations)
- [exclude combinations](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-excluding-configurations-from-a-matrix)